src/server/pfs/cmds/cmds_test.go

```diff
@@ -79,6 +80,24 @@ func TestPutFileSplit(t *testing.T) {
 	).Run())
 }
 
+func TestPutFileNonexistentRepo(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	repoName := tu.UniqueString("TestPutFileNonexistentRepo-repo")
+	start := time.Now()
+	require.NoError(t, tu.BashCmd(`
+                (dd if=/dev/urandom bs=1024 count=$((1024 * 1024)) 2>/dev/null | pachctl put file {{.repo}}@master:random 2>&1 || true)\
+                  | match "repo {{.repo}} not found"
+`,
+		"repo", repoName).Run())
+	// experimentally, the repo check should take less than three quarters
+	// of a second, while uploading 1GB of data takes well over two seconds
+	if time.Since(start) > 2*time.Second {
+		t.Error("pachctl put file appeared not to check if repo existed before uploading")
+	}
+}
```

Comment:
Maybe a more straightforward way to test this would be to use `pachctl put file -f does-not-exist, then make sure we get an error about the repo rather than the missing file?

Reply:
That _is_ dependent on the ordering of the two checks (repo versus file), but I like that it doesn’t rely on timing. I’m changing the test, and adding a comment so if the ordering changes the developer working on it has a hint about why the test failed.